File: python/project/weather.py
# weather program

# makes HTTP requests
import requests

# function to get weather data
def get_weather_data(city):
    api_key = '52da993b14bf13d19d1facc25fd0bca6'
    url = f'http://api.openweathermap.org/data/2.5/weather?q={city}&appid={api_key}&units=metric'
    response = requests.get(url)
    data = response.json()
    weather = data['weather'][0]['description']
    temperature = data['main']['temp']
    return weather, temperature

# main function
def main():
    city = input("Enter a city name: ")
    weather, temperature = get_weather_data(city) 
    print("Condition:",weather)
    print("Temperature: ",temperature)

main()

File: reactjs2/weather/Components/weather.js
import React, { useState } from 'react';

function Weather() {
  const [temperature, setTemperature] = useState('');
  const [description, setDescription] = useState('');
  const [location, setLocation] = useState('');

  const apiKey = '6e63f5d472671fed86266e9d95b72b1a';

  const handleFormSubmit = async (event) => {
    event.preventDefault();
    try {
      const response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${location}&units=metric&appid=${apiKey}`);
      const data = await response.json();
      setTemperature(data.main.temp);
      setDescription(data.weather[0].description);
    } catch (error) {
      console.error(error);
    }
  }

  const handleLocationChange = (event) => {
    setLocation(event.target.value);
  }

  return (
    <div className="weather-container">
      <h1 className="title">Weather App</h1>
      <form onSubmit={handleFormSubmit}>
        <input className="location-input" type="text" placeholder="Enter location" onChange={handleLocationChange} />
        <button className="submit-button" type="submit">Get Weather</button>
      </form>
      {temperature && description &&
        <div className="result-container">
          <p className="temperature">Temperature: {temperature} °C</p>
          <p className="description">Description: {description}</p>
        </div>
      }
    </div>
  );
}

export default Weather;
